mfr/extensions/image/templates/viewer.mako

```diff
@@ -1,4 +1,176 @@
+<script src="${base}/js/jquery.mousewheel.min.js"></script>
+<script src="${base}/js/jquery.zoom.js"></script>
+
+## The main script for enable Hi-Res and Zoom for rendered images
+<script>
+    ## Enforces strict mode
+    "use strict";
+
+    ## Magnifiation parameters
+    var magScale = 1;
+    var magStep = 0.1;
+    var maxScale = 3;
+    var minScale = 1;
+    var magHeight = null;
+    var magWidth = null;
+
+    ## The height and width for the displaying part of the image
+    var heightLimit = null;
+    var widthLimit = null;
+
+    ## The minimal height requirement to enable the zoom feature for a image
+    ## For more information, refer to the line where it is used in this file
+    var minHeightToZoom = 200;
+
+    ## A flag indicating whether the image is already wrapped or not
+    var wrapped = false;
+
+    $(document).ready(function() {
+
+        ## Reference on how mobile is detected: https://stackoverflow.com/a/10364620
+        var isMobile = window.matchMedia("only screen and (max-width: 760px)");
+
+        ## Enable the zoom feature only for desktop browsers
+        if (!isMobile.matches) {
+
+            ## Update the style for instruction popover and enable it
+            var popoverBuffer = $("#popover-buffer");
+            var popoverContent = $("#popover-content");
+            popoverBuffer.css({"height": "38px"});
+            popoverContent.css({
+                "position": "absolute",
+                "top": "10px",
+                "right": "10px",
+                "cursor": "pointer"
+            });
+            $('[data-toggle="popover"]').popover();
+
+            var baseImage = $("#base-image");
+
+            ## Quirks: the base image must be wrapped, see http://www.jacklmoore.com/zoom/
+            ##
+            ## The BASE image must be wrapped with a parent "container" and it must be the only (or
+            ## the first) image in it.  The `jquery.zoom` library performs the zoom on the parent
+            ## and grabs the first image it finds to create the zoom image.
+            ##
+            ## `addSpan()` performs the "wrap" accordingly to cater for most cases
+            var addSpan = function () {
+
+                ## Only wrap the image once
+                if (wrapped) {
+                    return;
+                }
+
+                ## Obtain the original height and width for the image loaded
+                var baseImageHeight = parseInt(baseImage.css("height"), 10);
+                var baseImageWidth = parseInt(baseImage.css("width"), 10);
+
+                ## Detect the Google Chrome browser
+                var isChrome = !!window.chrome;
+
+                ## Detect if the image is downsized due to screen size
+                var isActualSize = heightLimit === baseImageHeight || widthLimit === baseImageWidth;
+
+                if (isActualSize || !isChrome) {
+                    ## Use the default wrapping suggested by http://www.jacklmoore.com/zoom/ if
```

Comment:
Local tests verified that images with actual size displayed no longer need a second layer of wrapping. It now shares the same wrapping method with downsized but not-in-chrome images. Comments are updated.